test/formula.jl

```diff
@@ -365,4 +368,38 @@ module TestFormula
     mm = ModelMatrix(mf)
     @test mm.m[:, 2] == d[complete_cases(d), :x1m]
 
+## Promote non-redundant categorical terms to full rank
+
+d = DataFrame(x = rep([:a, :b], times = 4),
+              y = rep([:c, :d], times = 2, each = 2),
+              z = rep([:e, :f], each = 4))
+[pool!(d, name) for name in names(d)]
+cs = [name => SumContrasts for name in names(d)]
+d[:n] = 1.:8
+
+
+## No intercept
+mf = ModelFrame(n ~ 0 + x, d[1:2,:], contrasts=cs)
```

Comment:
I just didn't want to type out the additional rows of the model matrix.

I think I intended to test three-way and multiple two-way interactions (with different patterns of redundancy). I'm not sure that's necessary at this point, but there's some value in it at least from the perspective of being explicit about the expected behavior.